src/features/common/RedeemCode/EnterRedeemCode.tsx

```diff
@@ -0,0 +1,65 @@
+import TextField from '@mui/material/TextField';
+import { useTranslation } from 'next-i18next';
+import { useForm } from 'react-hook-form';
+import { ReactElement } from 'react';
+import CancelIcon from '../../../../public/assets/images/icons/CancelIcon';
+import styles from '../../../../src/features/user/Profile/styles/RedeemModal.module.scss';
+import Button from '@mui/material/Button';
+
+export interface EnterRedeemCodeProps {
+  setInputCode: React.Dispatch<React.SetStateAction<string | null>>;
+  inputCode: string | null;
+  changeRouteCode: () => void;
+  closeRedeem: () => void;
+}
+
+const EnterRedeemCode = ({
+  setInputCode,
+  inputCode,
+  changeRouteCode,
```

Comment:
`changeRouteCode` >> what does this do? Please give this a more understandable name.